firebase/firestore.rules

```diff
@@ -1,8 +1,164 @@
 rules_version = '2';
+
+function isAuthenticated() {
+  return request.auth.uid != null;
+}
+
+function isUser(id) {
+  return isAuthenticated() && request.auth.uid == id;
+}
+
+function isVerifiedCAV() {
+  // TODO: Actually use auth claims
+  return isAuthenticated() && true;
+}
+
+function isVerifiedPIN() {
+  // TODO: Actually use auth claims
+  return isAuthenticated() && true;
+}
+
+function currentUserIsCurrentPINRef(userID, database, data) {
+  return isAuthenticated() &&
+  ('pinUserRef' in data) &&
+  data.pinUserRef == /databases/$(database)/documents/users/$(userID)
+}
+
+function currentUserIsCurrentCAVRef(userID, database, data) {
+  return isAuthenticated() &&
+  ('cavUserRef' in data) &&
+  data.cavUserRef == /databases/$(database)/documents/users/$(userID)
+}
+
+function questionnaireBelongsToUser(userID, database, data) {
+    return isAuthenticated() &&
+           ('parentRef' in data) &&
+           data.parentRef == /databases/$(database)/documents/users/$(userID)
+}
+
+function confirmEitherPINorCAV(userID, database, data) {
+    return currentUserIsCurrentPINRef(userID, database, data) || currentUserIsCurrentCAVRef(userID, database, data)
+}
+
+function isNotSameCAVPIN(data) {
+  return data.cavUserRef != data.pinUserRef
+}
+
+function verifyUserDataComplete(data) {
+   return data.averageRating >= 1 &&
```

Comment:
Why 1? Won't a user start with rating =  0 ?

Reply:
The min rating you can receive is 1, but this brings up a good point... what's the default rating :D

Reply:
Updated this -- average rating should be nullable :)